src/Miningcore/Persistence/Postgres/Repositories/BalanceRepository.cs

```diff
@@ -116,6 +116,20 @@ public async Task<decimal> GetBalanceAsync(IDbConnection con, string poolId, str
 
             return await con.QuerySingleOrDefaultAsync<decimal>(query, new { poolId, address });
         }
+
+        public async Task<Balance> GetMinerBalanceAsync(IDbConnection con, string poolId, string address)
+        {
+            logger.LogInvoke();
+
+            const string query = "SELECT b.poolid, b.address, b.amount, b.created, b.updated, MAX(p.created) AS paiddate FROM balances AS b " +
```

Comment:
GROUP BY is not needed for an individual miner query. It can be something like this.

SELECT b.poolid, b.address, b.amount, b.created, b.updated, p.created AS paiddate FROM balances AS b 
LEFT JOIN payments AS p ON  p.address = b.address AND p.poolid = b.poolid 
WHERE b.poolid = @poolId AND b.address = @address ORDER BY p.created DESC LIMIT 1;